dateparser/conf.py

```diff
@@ -1,67 +1,72 @@
 # -*- coding: utf-8 -*-
 from pkgutil import get_data
 
+from itertools import chain
+from functools import wraps
 from yaml import load as load_yaml
 
-"""
-:mod:`dateparser`'s parsing behavior can be configured like below
+from .utils import Registry
 
-*``PREFER_DAY_OF_MONTH``* defaults to ``current`` and can have ``first`` and ``last`` as values::
 
-    >>> from dateparser.conf import settings
-    >>> from dateparser import parse
-    >>> parse(u'December 2015')
-    datetime.datetime(2015, 12, 16, 0, 0)
-    >>> settings.update('PREFER_DAY_OF_MONTH', 'last')
-    >>> parse(u'December 2015')
-    datetime.datetime(2015, 12, 31, 0, 0)
-    >>> settings.update('PREFER_DAY_OF_MONTH', 'first')
-    >>> parse(u'December 2015')
-    datetime.datetime(2015, 12, 1, 0, 0)
+class SettingsRegistry(Registry):
 
-*``PREFER_DATES_FROM``* defaults to ``current_period`` and can have ``past`` and ``future`` as values.
-Assuming current date is June 16, 2015::
+    @classmethod
+    def get_key(cls, *args, **kwargs):
+        if not args and not kwargs:
+            return 'default'
 
-    >>> from dateparser.conf import settings
-    >>> from dateparser import parse
-    >>> parse(u'March')
-    datetime.datetime(2015, 3, 16, 0, 0)
-    >>> settings.update('PREFER_DATES_FROM', 'future')
-    >>> parse(u'March')
-    datetime.datetime(2016, 3, 16, 0, 0)
+        keys= sorted(['%s-%s' % (key, str(kwargs[key])) for key in kwargs])
```

Comment:
Just curious, how hashing here would be of value for us? Given that `dict` automatically hashes the keys.